### ISSUE TITLE
Transaction (multi) for coroutine doesn't work

### ISSUE THREAD

**Issue by Distractic** [lettuce-io/lettuce-core#2371]
#### Current Behavior

Based on this connection pool : `BoundedAsyncPool<StatefulRedisConnection<ByteArray, ByteArray>>`

If we got a connection, use `sync()` method  and `multi` to make transaction, the transaction is performed as expected

```kt
val pool: BoundedAsyncPool<StatefulRedisConnection<ByteArray, ByteArray>> = ...
val connection: StatefulRedisConnection<ByteArray, ByteArray> = pool.acquire().await()
val result = connection.sync().multi {
   get(key1)
   get(key2)
   sismember(key3, "test".encodeToByteArray())
}

println("Executed multi: $result ") // executed
println("Result 0: ${result.get<ByteArray>(0)}") // executed
```

The prints display information as expected and the transaction is executed.

However, if we're using the coroutine, the transaction will just block and never be performed
```kt
val pool: BoundedAsyncPool<StatefulRedisConnection<ByteArray, ByteArray>> = ...
val connection: RedisCoroutinesCommands<ByteArray, ByteArray> = pool.acquire().await().coroutines()
val result = connection.multi {
   get(key1)
   get(key2)
   sismember(key3, "test".encodeToByteArray())
}

println("Executed multi: $result ") // never executed
println("Result 0: ${result.get<ByteArray>(0)}") // never executed
```
<details>
<summary>Stack trace</summary>

```java
Nothing thrown.
```

</details>

#### Input Code

[Repository](https://github.com/Rushyverse/core/blob/feat/guild/src/main/kotlin/com/github/rushyverse/core/data/Guild.kt)

<details>

</details>

#### Expected behavior/code

The transaction should  be executed using coroutine extension

#### Environment

- Lettuce version(s): [6.2.3.RELEASE]
- Redis version: [7.0.8]
- Kotlin: [1.8.0]
- Coroutine: [1.6.4]

**Comment by mp911de** [comment]
Looks like a duplicate of #1954.

We deprecated the `multi` extension because of that reason as Redis' transaction do not work well with an async API.

**Comment by Distractic** [comment]
Ok thanks ! And sorry for the duplicate 

So for the moment, we can't use transaction. I will just make several requests in this case while waiting for a possible fix.

**Comment by mp911de** [comment]
Yes, sorry for the inconvenience. Let me reopen the ticket so we add proper deprecation annotations.

**Comment by young0264** [comment]
Hi @mp911de,

I checked the current coroutine transaction API.

It looks like the DSL-style RedisCoroutinesCommands.multi { ... } extension is already documented as deprecated since 6.1.6, but it is missing a Kotlin @Deprecated annotation, so users do not get compiler or IDE warnings.

The issue seems to come from the mismatch between Redis transaction semantics and the coroutine suspend command model: commands issued after MULTI are queued and their actual results are only available after EXEC, while the DSL makes it easy for users to write code that awaits those command results inside the transaction block.

I opened a small PR that adds the missing @Deprecated annotation to that extension only, without changing runtime behavior.

For the deprecation message, I used a shorter message to keep it closer to the existing Lettuce style:
@Deprecated("since 6.1.6, coroutine transaction DSL commands complete only after EXEC")

Thank you.
